python/ray/data/_internal/logical/operators/read_operator.py

```diff
@@ -93,3 +93,12 @@ def _cached_output_metadata(self) -> BlockMetadata:
             input_files=input_files,
             exec_stats=None,
         )
+
+    @classmethod
+    def is_read_op(cls):
+        return True
+
+    def can_modify_num_rows(self) -> bool:
+        # NOTE: Returns true, since most of the readers expands its input
+        #       and produce many rows for every single row of the input
+        return True
```

Comment:
(can handle this in the next PR) it'd be better to decide this based on the data source type. 
e.g., image data source won't modify num rows

Reply:
Yeah, i thought about doing that but then ultimately decided not to (for a reason that i can't recollect now).

Let me do that in a follow-up.